src/bindings.ts

```diff
@@ -54,7 +54,17 @@ export function bindRoutes(routerRoutes: any, controllers: any[], getter?: (ctrl
 
         const args = getArguments(params, ctx, next);
         const result = inst[name](...args);
-        if(result) ctx.body = await result;
+        if(result) {
+          const body = await result;
+          const fileDownload = body as FileDownload;
+          if(fileDownload) {
```

Comment:
you can use an `instanceOf` check here instead: `if (body instanceof FileDownload) {`

Example:
```
class Foo {
  bar: boolean;
}

const foo: Foo = new Foo();
foo.bar = true;

console.log(foo instanceof Foo);
// -> true
```

Reply:
except since I am using fileDownload, it is faster and more efficient to cast first and then check for null

Reply:
How is an assignment and casting faster than a simple type check?
Also, take a look at what happens when the code is transpiled to JS:
```javascript
if (result) {
    const body = await result;
    const fileDownload = body;
    if (fileDownload) {
        ctx.res.setHeader('Content-type', fileDownload.mimeType);
        ctx.res.setHeader('Content-disposition', ('attachment; filename=' + fileDownload.fileName));
        ctx.body = fileDownload.stream;
    }
    else {
        ctx.body = body;
    }
}
```
See how your version doesn't result in any actual type checking?

Reply:
Interesting I assumed it transpiled to something smart. This method is the most efficient method in Python and c# if you have to cast it to the type you need anyway, to get to the class properties. Apparently JS doesn't care, that there will be properties not mapped at all.